### DISCUSSION TITLE
Copy/move file between sites

### DISCUSSION THREAD

**Discussion by wonderplayer** [pnp/pnpcore#406] [Q&A] [answered]
Hello,
Is it possible to copy/move file between different sites? I can do it with Power Automate flow, but with PnPCore and raw REST API I can't seem to figure out the way how to do it.
I noticed that there is some logic [here](https://github.com/pnp/pnpcore/blob/3bc2ccfab4efe067e7cde2f07e9f65ec30bf588c/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/File.cs#L521) to check, whether destination web is the same as source web, but this doesn't help.
Tried to execute request with Postman, which is being executed by SDK, and found out, that I get following response
![image](https://user-images.githubusercontent.com/15668343/115568235-c3f78380-a2c4-11eb-8dbc-2c8307c96e82.png)
I'm authorizing with an Azure app, that has Sites.FullControl.All permissions for SharePoint.

So... is that a bug, or I'm doing something incorrect here?

**Comment by wonderplayer** [comment, marked as answer]
Was trying to find more info. Seems that this action can't handle different site collection?

- [Example within one site collection](https://gist.github.com/phillipharding/7031a510ce76c72d41b653076ff838a1)
- [Example with comment about move to different site collection](https://gist.github.com/zplume/9f4c1a658517802701deff3473f23a60) (see first comment)

Guess then I'm left with an option to download the file and upload the file using PnPCore, as PnPCore is using SP.MoveCopyUtil.CopyFileByPath()?

On the other hand, [this](https://gist.github.com/zplume/21248c3a8a5f840a366722442cf9ee97) resource suggests that it is possible with SP.MoveCopyUtil...

**Reply by wonderplayer** [reply]
Achieved file copy between site collections using [this](https://gist.github.com/zplume/21248c3a8a5f840a366722442cf9ee97) post. There were no issues with permissions.
Looks like this is using underlying Microsoft 365 Copy and Move API.

**Reply by jansenbe** [reply]
@wonderplayer : Cross site collection copy/move is not supported in PnP Core SDK at this moment, but it's doable via REST as shown by PnP PowerShell: https://github.com/pnp/powershell/blob/dev/src/Commands/Utilities/CopyMover.cs. I'll add this request to our backlog. Thanks for making us aware.
